testing-cci-datasets/generate_summary.py

```diff
@@ -247,68 +182,32 @@ def main():
     if len(sys.argv) == 3:
         test_mode = sys.argv[2]
 
-    support_file_name = f'{date_today}_test_{store_name}_data_support'
     results_dir = f'{store_name}'
     if test_mode:
         results_dir = f'{test_mode}/{store_name}'
+
+    support_file_name = f'{date_today}_test_{store_name}_data_support'
     results_csv = f'{results_dir}/{support_file_name}.csv'
-    sort_csv(results_csv, f'{results_dir}/{support_file_name}_sorted.csv')
 
-    test_data_sets = read_all_result_rows(
-        f'{results_dir}/{support_file_name}_sorted.csv', header_row)
+    results_sorted = sort_csv(results_csv)
+    test_data_sets = read_all_result_rows(results_sorted, header_row)
 
     ecvs = get_list_of_ecvs(test_data_sets)
-    failed_csv = f'{results_dir}/{support_file_name}_failed.csv'
-    create_list_of_failed(test_data_sets, failed_csv, header_row)
-    if os.path.exists(failed_csv):
-        sort_csv(failed_csv,
-                 f'{results_dir}/{support_file_name}_sorted_failed.csv')
-
-    with open(results_csv, 'r', newline='') as f_input:
-        csv_input = csv.DictReader(f_input)
-        data = sorted(csv_input, key=lambda row: (row['Dataset-ID']))
-    with open(f'{results_dir}/{support_file_name}_sorted.csv', 'w',
-              newline='') as f_output:
-        csv_output = csv.DictWriter(f_output, fieldnames=csv_input.fieldnames)
-        csv_output.writeheader()
-        csv_output.writerows(data)
+    # below maybe not needed anymore?
+    # failed_csv = f'{results_dir}/{support_file_name}_failed.csv'
+    # failed_sorted = create_list_of_failed(test_data_sets, failed_csv,
+    #                                       header_row)
 
     summary_csv = f'{results_dir}/{support_file_name}_summary_sorted.csv'
-    header_summary = ['ecv', 'supported', 'open_success', 'open_fail',
-                      'open_temp_success',
-                      'open_temp_fail', 'open_bbox_success', 'open_bbox_fail',
-                      'cache_success',
-                      'cache_fail', 'visualize_success', 'visualize_fail',
-                      'supported_percentage',
-                      'open_success_percentage', 'open_temp_success_percentage',
-                      'open_bbox_success_percentage',
-                      'cache_success_percentage',
-                      'visualize_success_percentage',
-                      'total_number_of_datasets']
+    header_summary = None
 
     for ecv in ecvs:
         results_summary_row = count_success_fail(test_data_sets, ecv)
+        if not header_summary:
+            header_summary = list(results_summary_row.keys())
```

Comment:
Otherwise it is confusing to read.

```suggestion
    for ecv in ecvs:
        results_summary_row = count_success_fail(test_data_sets, ecv)
        header_summary = list(results_summary_row.keys())
```